src/Factory/BaseFactory.php

```diff
@@ -609,26 +609,26 @@ public function skipSetterFor($skippedSetters, bool $merge = false)
      * Query the factory's related table without before find.
      *
      * @param string $type the type of query to perform
-     * @param array $options An array that will be passed to Query::applyOptions()
-     * @return \Cake\ORM\Query The query builder
-     * @see Query::find()
+     * @param mixed ...$options Options passed to the finder
+     * @return \Cake\ORM\Query\SelectQuery The query builder
+     * @see \Cake\ORM\Query\SelectQuery::find()
      */
-    public static function find(string $type = 'all', array $options = []): Query
+    public static function find(string $type = 'all', mixed ...$options): SelectQuery
     {
-        return (new static())->getTable()->find($type, $options);
+        return (new static())->getTable()->find($type, ...$options);
     }
 
     /**
      * Get from primary key the factory's related table entries, without before find.
      *
      * @param mixed $primaryKey primary key value to find
-     * @param array $options options accepted by `Table::find()`
+     * @param mixed ...$options options accepted by `Table::get()`
      * @return \Cake\Datasource\EntityInterface
      * @see Table::get()
      */
-    public static function get($primaryKey, array $options = []): EntityInterface
+    public static function get(mixed $primaryKey, mixed ...$options): EntityInterface
```

Comment:
The signature of Table::get() has changed significantly in Cake5. I would suggest to take the exact same signature. Otherwise we get some tets marked as deprecated.